### PR TITLE
Windows build scripts (NR-52407)

### PR DESCRIPTION
**Pull request by kang-makes**
I have changed the `build.sh` to `build-linux.sh` until we find the way we want to refactor the pipeline of this repository. Then I added `build-windows.sh` scripts to build windows binaries.

I made them work in the forked repository: https://github.com/newrelic-forks/newrelic-prometheus-exporters-packages/tree/d350c05b79ca242da30b3144e54c37a0f6b3ab86

For the sake of not forgetting what happened in the fork repository (as logs can rotate), here are the [logs of the workflow run](https://github.com/newrelic-forks/newrelic-prometheus-exporters-packages/actions/runs/3497378029/jobs/5856406422):
```
   creating: IBM-MQC-Redist-Win64/zips/
  inflating: IBM-MQC-Redist-Win64/zips/UnpackFeature.cmd  
  inflating: IBM-MQC-Redist-Win64/zips/UnpackFeature.ps1  
++ cygpath -aw .
+ export 'CGO_CFLAGS=-IC:\Users\runneradmin\AppData\Local\Temp\tmp.sVtijfM2aF\IBM-MQC-Redist-Win64\tools\c\include -D_WIN64'
+ CGO_CFLAGS='-IC:\Users\runneradmin\AppData\Local\Temp\tmp.sVtijfM2aF\IBM-MQC-Redist-Win64\tools\c\include -D_WIN64'
++ cygpath -aw .
+ export 'CGO_LDFLAGS=-LC:\Users\runneradmin\AppData\Local\Temp\tmp.sVtijfM2aF\IBM-MQC-Redist-Win64\bin64 -lmqm'
+ CGO_LDFLAGS='-LC:\Users\runneradmin\AppData\Local\Temp\tmp.sVtijfM2aF\IBM-MQC-Redist-Win64\bin64 -lmqm'
++ date +%Y%m%d-%H%M%S
++ uname -s
++ uname -i
++ git rev-list -1 HEAD --abbrev-commit
+ go build '-ldflags=-X "main.BuildStamp=20221118-134508" -X "main.BuildPlatform=MINGW64_NT-10.0-20348/unknown" -X "main.GitCommit=d268624"' -o ibmmq-exporter.exe '.\cmd\mq_prometheus'
+ mkdir -p D:/a/newrelic-prometheus-exporters-packages/newrelic-prometheus-exporters-packages/src/repo/exporters/ibmmq/target/bin
+ cp ibmmq-exporter.exe D:/a/newrelic-prometheus-exporters-packages/newrelic-prometheus-exporters-packages/src/repo/exporters/ibmmq/target/bin/ibmmq-exporter.exe
Using Default prometheus template
mingw32-make[1]: Entering directory 'D:/a/newrelic-prometheus-exporters-packages/newrelic-prometheus-exporters-packages/src/repo/nri-config-generator'
go build \
-ldflags "-X main.integration=nri-ibmmq -X main.integrationVersion=0.3.0 -X main.gitCommit=d350c05b79ca242da30b3144e54c37a0f6b3ab86 -X main.buildDate=2022-11-18_01:46:11PM" \
-o D:/a/newrelic-prometheus-exporters-packages/newrelic-prometheus-exporters-packages/src/repo/exporters/ibmmq/target/bin/nri-ibmmq github.com/newrelic/nri-config-generator
go: downloading github.com/Masterminds/sprig/v3 v3.2.2
[...]
go: downloading github.com/pelletier/go-toml v1.9.5
mingw32-make[1]: Leaving directory 'D:/a/newrelic-prometheus-exporters-packages/newrelic-prometheus-exporters-packages/src/repo/nri-config-generator'
executable file was created D:/a/newrelic-prometheus-exporters-packages/newrelic-prometheus-exporters-packages/src/repo/exporters/ibmmq/target/bin/nri-ibmmq
```

For more context on why we are creating a script to compile IBMMQ on Windows:
 - The batch script does not load the shared libraries needed and has no variables to change it: https://github.com/ibm-messaging/mq-metric-samples/blob/master/scripts/buildMonitors.bat#L21
 - The batch script IBM provides compiles ALL the metric providers: https://github.com/ibm-messaging/mq-metric-samples/blob/master/scripts/buildMonitors.bat#L25-L41

So the binary compiles but the pipeline still failed as there is no packaging step working yet (and the fork lacks of the secrets needed to package things).

The work on this task ends here as there this has to integrate with the Wix project and glue everything together.

### CODE REVIEW

**Review by sigilioso — APPROVED** [review]
LGTM 👍 

Thanks not only for taking care of the task, but also getting rid of PowerShell scripts and cleaning up the pipelines a bit.

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -74,7 +71,7 @@ jobs:
       UPLOAD_URL: ${{ steps.create_release.outputs.upload_url }}
       INTEGRATION_NAME: ${{ steps.check.outputs.NAME }}
       INTEGRATION_VERSION: ${{ steps.check.outputs.VERSION }}
-      RELEASE_TAG: ${{ steps.check.outputs.PACKAGE_NAME }}-${{ steps.check.outputs.VERSION }}
+      RELEASE_TAG: ${{ steps.check.outputs.PACKAGE_NAME }}-${{ steps.check.outputs.VERSION }}q
```
Comment (marcsanmi) [inline .github/workflows/prerelease.yml:74]: Is this trailing `q` expected? 🤔

Comment (kang-makes) [reply .github/workflows/prerelease.yml:74]: Oops. I think it was a vim `:q` typo. Fixing it.

Comment (kang-makes) [reply .github/workflows/prerelease.yml]: Fixed in [#3d140d3](https://github.com/newrelic/newrelic-prometheus-exporters-packages/pull/133/commits/3d140d3fb93adf5e75ba32a64088c1b656641490)

**Pull request merged**